src/components/pages/pages.test.tsx

```diff
@@ -0,0 +1,178 @@
+  const moveToPreviousPage = () => setPageIndex(pageIndex - 1);
+
+  return (
+    <Pages pageIndex={pageIndex} {...props}>
+      <Page
+        title={
+          <Heading
+            data-element="firstHeader"
+            title="My First Page"
+            backLink={moveToPreviousPage}
+          />
+        }
+      >
+        <Button onClick={moveToNextPage}>Go to second page</Button>
+      </Page>
+      <Page
+        title={
+          <Heading
+            data-element="secondHeader"
+            title="My Second Page"
+            backLink={moveToPreviousPage}
+          />
+        }
+      >
+        Second page
+      </Page>
+      <Page
+        title={
+          <Heading
+            data-element="thirdHeader"
+            title="My Third Page"
+            backLink={moveToPreviousPage}
+          />
+        }
+      >
+        Third page
+        <Button onClick={moveToNextPage}>Go to next page</Button>
+      </Page>
+    </Pages>
+  );
+};
+
+beforeAll(() => {
+  jest.useFakeTimers();
+});
+
+afterAll(() => {
+  jest.useRealTimers();
+});
+
+test.each([
+  ["fade", "carousel-transition-fade"],
+  ["slide", "slide-next"],
+])(
+  "should render correct animation when the `transition` prop is `%s`",
+  async (transition, expected) => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(<PagesExample transition={transition} />);
+
+    await user.click(screen.getByRole("button", { name: "Go to second page" }));
+    jest.runAllTimers();
+
+    expect(screen.getByTestId("visible-page")).toHaveClass(
+      `${expected}-enter-done`
+    );
+  }
+);
+
+test("when the `pageIndex` prop is greater than the number of `Page` children, the first page is rendered", () => {
+  render(
+    <Pages pageIndex={100}>
+      <Page title={<Heading title="Page 1" />}>First Page</Page>
+      <Page title={<Heading title="Page 2" />}>Second Page</Page>
+      <Page title={<Heading title="Page 3" />}>Third Page</Page>
+    </Pages>
+  );
+
+  expect(screen.getByRole("heading")).toHaveTextContent("Page 1");
+  expect(screen.getByTestId("visible-page")).toHaveTextContent("First Page");
+});
+
+test("when the `pageIndex` prop is changed to `undefined`, the currently-rendered page does not change", () => {
+  const ExampleComponent = ({ pageIndex }: Pick<PagesProps, "pageIndex">) => (
+    <Pages pageIndex={pageIndex}>
+      <Page title={<Heading title="Page 1" />}>First Page</Page>
+      <Page title={<Heading title="Page 2" />}>Second Page</Page>
+      <Page title={<Heading title="Page 3" />}>Third Page</Page>
+    </Pages>
+  );
+  const { rerender } = render(<ExampleComponent pageIndex={1} />);
+  expect(screen.getByRole("heading")).toHaveTextContent("Page 2");
+
+  rerender(<ExampleComponent pageIndex={undefined} />);
+  jest.runAllTimers();
+
+  expect(screen.getByRole("heading")).toHaveTextContent("Page 2");
+});
+
+test("navigating to the next page should work as expected", async () => {
```

Comment:
**suggestion(non-blocking)**: for this test and the succeeding one, what do you think about this so the expected behaviour is clearer? 

```suggestion
test("navigating to the next page renders its content", async () => {
```